### PR TITLE
Add carlos

### PR DESCRIPTION
**Pull request by cayorodriguez**
Changes in catalan tokenization (especially enclitic pronouns and contractions) and overall lemmatization for better processing and to align with changes in UD version 3.8. Will do a separate PR for lookups.

## Description

Changed punctuation.py, __init__.py, tokenizer exceptions

### Types of change

minor enhancements, no new functionalities

## Checklist

- [ ] I have submitted the spaCy Contributor Agreement. <!-- Waiting for changes to DCO
- [ X] I ran the tests, and all new and existing tests passed.
- [X ] My changes don't require a change to the documentation, or if they do, I've added all required information.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -124,3 +124,5 @@ Desktop.ini
 
 # IPython
 .ipynb_checkpoints/
+
+>>>>>>> Stashed changes
```
Comment (svlandeg) [inline .gitignore:128]:
```suggestion
```

**Review by svlandeg — COMMENTED** [review]
Hi! 

I saw that you requested a review, but could you address my earlier comments first, as well as the failing test? Ideally, the CI run should pass (all green) for the entire PR.

**Comment by cayorodriguez** [comment]
I  have done all changes, merges, etc. requested, and tested on two ubuntu
machines without issues (except when using the --slow parameter in pytest
that includes classification tests). I don't think these failures relate to
our changes.

Please review again. We still need to prepare our PR for the lookup files
for the lemmatizer.
==============
Carlos Rodriguez
==============

**Comment by svlandeg** [comment]
Hi, 

maybe you have uncommited changes, because the edits I requested for the `Makefile` and `requirements.txt` weren't commited, even though you marked the conversations as resolved. I've unresolved them for now to keep a clear overview of this PR.

I do think the failing CI is directly related to the changes in this PR as well: `tests/lang/ca/test_exception.py::test_ca_tokenizer_handles_exc_in_text` is failing.

**Comment by cayorodriguez** [comment]
Sorry about this, will review the changes and commits, since that test is
not failing on my tests.

==============
Carlos Rodriguez
==============

**Comment by svlandeg** [comment]
Hi,

Thanks for hanging in there with us, this is starting to look almost finished and it's great to have the CI go green :-)

When I look at the files that were changed now in the updated version of the PR though, it looks like again some commits from other PRs have crept in (e.g the stuff from spancategorizer and universe.json). Could you update the base branch to make sure only your edits are present in this PR? I fear that otherwise the history might look a bit weird.

**Comment by cayorodriguez** [comment]
Reverting creeping commits now

**Comment by cayorodriguez** [comment]
Found a small inconsistency in the tokenization. Will correct it today, and commit later

File: .gitignore [hunk 2]
```diff
@@ -124,3 +124,4 @@ Desktop.ini
 
 # IPython
 .ipynb_checkpoints/
+
```
Comment (svlandeg) [inline .gitignore:127]:
```suggestion
```

File: Makefile [hunk 3]
```diff
@@ -5,7 +5,7 @@ override SPACY_EXTRAS = spacy-lookups-data==1.0.2 jieba spacy-pkuseg==0.0.28 sud
 endif
 
 ifndef PYVER
-override PYVER = 3.6
+override PYVER = 3.8
```
Comment (svlandeg) [inline Makefile:8]:
```suggestion
override PYVER = 3.6
```

File: .gitignore (hunk 2 above)
Comment (svlandeg) [reply .gitignore:127]: You should be able to just commit this suggestion (I don't have write access to your branch)

File: Makefile (hunk 3 above)
Comment (svlandeg) [reply Makefile:8]: You should be able to just commit this suggestion (I don't have write access to your branch)

**Comment by cayorodriguez** [comment]
Fixed!
==============
Carlos Rodriguez
==============

**Review by svlandeg — APPROVED** [review]
Great, thanks, this is looking good. 

My colleague Adriane will probably want to review this once more next week, but in the meantime feel free to start working on the lookups PR as well!

**Comment by adrianeboyd** [comment]
Hi, sorry for the delay!

I was hoping some recent updates to `develop` would clear up the remaining unrelated commits, but it didn't quite work and we don't have permission to push to your branch, so instead I created a new branch / PR that preserves all your work and cleans up the remaining (very minor) issues. Please see: #9297

I evaluated it on the v2.8 AnCora-Catalan release and the tokenization is at 99.97. If this is what you're expecting, we can get that merged and then finish testing the lemma data PR.

**Comment by cayorodriguez** [comment]
Hi Adriane,

This is what we expect, yes. Thanks
==============
Carlos Rodriguez
==============

**Comment by adrianeboyd** [comment]
Good, thanks! I've merged the linked PR, so I'll close this one now.

**Pull request closed without merging**